addons/pos_self_order/static/src/app/components/product_card/product_card.js

```diff
@@ -93,7 +93,32 @@ export class ProductCard extends Component {
         }
 
         if (product.isCombo()) {
-            this.router.navigate("combo_selection", { id: product.id });
+            const selectedCombos = [];
+
+            const showComboSelectionPage = product.combo_ids.some(({ combo_line_ids }) => {
+                if (combo_line_ids.length > 1 || combo_line_ids[0]?.product_id.isConfigurable()) {
+                    return true;
+                }
+                selectedCombos.push({
+                    combo_line_id: this.selfOrder.models["pos.combo.line"].get(
+                        combo_line_ids[0].id
+                    ),
+                    configuration: {
+                        attribute_custom_values: [],
+                        attribute_value_ids: [],
+                        price_extra: 0,
+                    },
+                });
+                return false;
+            });
```

Comment:
```suggestion
            const selectedCombos = [];
            let showComboSelectionPage = false;

            for (const combo of product.combo_ids) {
                const { combo_line_ids } = combo;
                if (combo_line_ids.length > 1 || combo_line_ids[0]?.product_id.isConfigurable()) {
                    showComboSelectionPage = true;
                    break;
                }
                selectedCombos.push({
                    combo_line_id: this.selfOrder.models["pos.combo.line"].get(
                        combo_line_ids[0].id
                    ),
                    configuration: {
                        attribute_custom_values: [],
                        attribute_value_ids: [],
                        price_extra: 0,
                    },
                });
            }
```

Rewrite using a normal for-loop. Using `.some` to do for-loop logic is confusing. It's not easy to know if it breaks at some point or not.

Reply:
Hello @caburj, changes are done, please check.
Thanks!